addOns/spider/src/main/java/org/zaproxy/addon/spider/ExtensionSpider2.java

```diff
@@ -77,7 +202,734 @@ public String getUIName() {
         return Constant.messages.getString("spider.addon.name");
     }
 
-    private ExtensionSpider getExtensionSpider() {
-        return Control.getSingleton().getExtensionLoader().getExtension(ExtensionSpider.class);
+    private ExtensionSpider2 getExtensionSpider() {
```

Comment:
It needs  to keep `ExtensionSpider`.

Reply:
IIRC that's where things start to fall apart.

Reply:
It is because then at hook it has to use the old class when the core isn't disabled. Then the parsers and filters all go to hell.

Reply:
Could you be more clear what's the problem? If core is not yet deprecated it needs to continue adding the `SvgHrefSpider` to core extension not the one in the add-on.

Reply:
Sure at this point: `spider.addCustomParser(svgHrefSpider);`  it needs to add a parser but that parser is based on the object in the add-on not in the core, but obviously the core is looking for the core object type.

Reply:
Got it, I thought it was still extending the core class but either way it will lead to the same problem. We need to add an adapter class.

Reply:
I tried reading about that, but I don't see how it will return the actually functionality. I get that you can build similar objects with the local vars and stuff but I don't get how to pass/maintain functionality.

Reply:
Or, which objects get used in which places...

Reply:
Pushed a commit which should address this case. It contains other tweaks that should be added to the first commit probably.
Feel free to remove the commit once you checked it.

Reply:
TY!

Reply:
Thanks very much @thc202. Do you want me to add a co-authored line to the first commit (where I added your patch)?

Reply:
No.